<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Getting Started</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Home</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="ChIP-seq_workshop_intro.html">Getting Started</a>
</li>
<li>
  <a href="ChIP-seq_workshop_QC.html">Quality Control</a>
</li>
<li>
  <a href="ChIP-seq_workshop_map.html">Mapping and Filtering</a>
</li>
<li>
  <a href="ChIP-seq_workshop_visualisation.html">Visualisation</a>
</li>
<li>
  <a href="ChIP-seq_workshop_analysis.html">Analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Getting Started</h1>

</div>


<head>
<script src="https://kit.fontawesome.com/ece750edd7.js" crossorigin="anonymous"></script>
</head>
<hr />
<div id="getting-started-with-linux" class="section level3">
<h3>Getting Started with Linux</h3>
<p>We are using the Linux command line to run most of the tools we use today. If you are new to Linux please complete the Intro to Command Line <a href="http://bifx-core.bio.ed.ac.uk/hyweldd/training/Bioinformatics_on_the_Command_Line/">Workshop</a>.</p>
<p><br></p>
</div>
<div id="logging-in" class="section level3">
<h3>Logging in</h3>
<p>We have several servers that you can login to. For the purpose of this practical we will use bifx-rta. No matter where you login, you will have access to the same files and programs. bifx-core2 is also available, both bifx-rta and bifx-core2 require you to use a VPN.</p>
<p>There are several options to login to our machines. You can use the <strong>Terminal</strong> app on a Mac or equivalant <strong>Command Prompt</strong> or MobaXTerm on Windows. Login via <strong>X2G0</strong> if you want a graphical interface.</p>
<p>To login via command line: ssh <a href="mailto:USER@bifx-rta.bio.ed.ac.uk" class="email">USER@bifx-rta.bio.ed.ac.uk</a></p>
<p>Login with your user credentials for our servers (not the same as EASE)</p>
<p>If you are using MobaXTerm, an alternative way of logging in to the server is shown in the MobaXTerm demo.</p>
<p>Once you have typed in your password, you should see some welcome text and a prompt that looks something like this:</p>
<p>[USERNAME]<span class="citation">@bifx-rta</span>:~$</p>
<!--Alternatively you can use **RStudio** which is an environment built for R programming but also gives you access to the command line terminal, your file structure and several other graphical features. 

1. Open this link in a new window: [bifx-rta.bio.ed.ac.uk](http://bifx-rta.bio.ed.ac.uk:8787)
2. Login with your user credentials for our servers (not the same as EASE)
3. Familiarise yourself with the different panels and locate the Terminal. -->
<p><br></p>
</div>
<div id="creating-a-web-directory" class="section level3">
<h3>1. Creating A Web Directory</h3>
<p>In order to view files created on the server, we need to create a public_html directory.</p>
<p>After logging in you should be in your $HOME directory, check with;</p>
<pre class="bash"><code>pwd</code></pre>
<p>This should show the PATH of your present working directory, which should now be your home directory as you have just logged in. You can return to this place at any time using the change directory command.</p>
<pre class="bash"><code>cd</code></pre>
<p>You have permissions to create files and directories under your home folder. Lets create some now which we will use later on.</p>
<pre class="bash"><code>mkdir ~/public_html
mkdir ~/public_html/TMP</code></pre>
<p>Here we have used the absolute path name for each directory using ~/ as a shortcut for your $HOME directory. Nested directories are separated by the forward slash ‘/’ sign.</p>
<p>As you have created ~/public_html, contents of this directory are available online with any web browser</p>
<p>To see it enter the following URL, changing yourUserName to what ever your username is.</p>
<p><a href="http://bifx-core.bio.ed.ac.uk/~yourUserName" class="uri">http://bifx-core.bio.ed.ac.uk/~yourUserName</a></p>
<p><br></p>
</div>
<div id="chip-seq-sequencing-data" class="section level3">
<h3>2. ChIP-seq sequencing data</h3>
<p>The datasets used in this exercise are derived from a single end ChIP-seq experiment (actually ChIP-exo) in <em>S.cerevisiae</em>. There are 2 biological replicates (though we recommend using 3 or more!) Reb1_R1 and Reb1_R2 as well as their corresponding input controls Input_R1 and Input_R2. For this experiment immunoprecipitation was performed with antibodies against Reb1. Reb1 recognizes a specific sequence (TTACCCG) and is involved in many aspects of transcriptional regulation by all three yeast RNA polymerases and promotes formation of nucleosome-free regions (NFRs). You can find the original publication <a href="http://www.sciencedirect.com/science/article/pii/S0092867411013511">here</a>. For the purpose of this workshop we have randomly subsampled and filtered out poor quality reads to speed up runtime.</p>
<table>
<thead>
<tr class="header">
<th>Dataset</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Reb1_R1</td>
<td>ChIP experiment, replicate 1</td>
</tr>
<tr class="even">
<td>Reb1_R2</td>
<td>ChIP experiment, replicate 2</td>
</tr>
<tr class="odd">
<td>Input_R1</td>
<td>Input DNA, replicate 1</td>
</tr>
<tr class="even">
<td>Input_R2</td>
<td>Input DNA, replicate 2</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
<div id="obtaining-data" class="section level3">
<h3>Obtaining data</h3>
<p>First, make a new directory for this tutorial and move into that directory. Then link the directory to your public html folder as we are going to make everything public in this tutorial.</p>
<pre class="bash"><code>cd 
mkdir ChIP-seq_workshop
cd ChIP-seq_workshop
ln -s $PWD ~/public_html/</code></pre>
<p>Next, create a subfolder called <strong>fastq</strong> for all of our sequence files and link the raw datasets to this folder:</p>
<pre class="bash"><code>mkdir fastq
cp /homes/library/training/ChIP-seq_workshop/data/*fq.gz fastq/.</code></pre>
<p>When you receive data from a sequencing centre the file should also be provided with an alphanumeric string known as an <strong>md5 checksum</strong>. We can think of this as a files passport or fingerprint and use it to verify our data and ensure it wasn’t corrupted or truncated during download. The md5 checksums for these files are below. Lets check that now using the <code>md5sum</code> command:</p>
<table>
<thead>
<tr class="header">
<th>md5 checksum</th>
<th>filename</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>914b4dda687a76b0d50e545e3ce705d6</td>
<td>Input_R1.fq.gz</td>
</tr>
<tr class="even">
<td>f421ed18b71a801b236612cdde49dbaf</td>
<td>Input_R2.fq.gz</td>
</tr>
<tr class="odd">
<td>dd363301ad237ecb6c81c59ae97995a2</td>
<td>Reb1_R1.fq.gz</td>
</tr>
<tr class="even">
<td>06623f9e556876dd6c4d1dfdc4348698</td>
<td>Reb1_R2.fq.gz</td>
</tr>
</tbody>
</table>
<pre class="bash"><code>cd fastq #Move into the fastq directory
md5sum *.fq.gz &gt; md5
cat md5 #prints out the contents of md5
#To check the files and md5 sums match at any time
md5sum -c md5 </code></pre>
<p><br></p>
</div>
<div id="integrative-genomics-viewer" class="section level3">
<h3>Integrative Genomics Viewer</h3>
<p>Please install <a href="https://software.broadinstitute.org/software/igv/download">IGV</a> on your own machine, alternatively you can use the <a href="https://igv.org/app/">App</a>.</p>
<div class="key-points">
<h2>
<i class="fas fa-thumbtack"></i>Key Aims:
</h2>
<div id="section" class="section level2">
<h2></h2>
<ul>
<li>Be able to login to the bifx servers</li>
<li>Create a personal web directory</li>
<li>Create a project directory for fastq files</li>
<li>Install IGV on your own machine</li>
</ul>
</div>
</div>
<p><br></p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
